### PR TITLE
Add Require Inspector extension

### PR DESCRIPTION
**Pull request by zeroZshadow**
#### Description
Added an extension to the monobehaviour inspector to visuals `[Require]` states.
Added `io.improbable.gdk.debug` package to house functionality.

#### Tests
Tested in Playmode and Edit mode

#### Documentation
- [ ] Changelog
- [ ] Docs under Require flow documentation

image

### CODE REVIEW

**Comment by zeroZshadow** [comment]
I've noticed I still need to clean a few test things, but wanted to make the PR

File: workers/unity/Assets/Playground/Scripts/MonoBehaviours/WorkerFlagTest.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 namespace Playground
 {
-    [WorkerType(WorkerUtils.UnityGameLogic)]
+    [WorkerType(WorkerUtils.UnityGameLogic, "FakeWorkerType")]
```
Comment (jamiebrynes7) [inline workers/unity/Assets/Playground/Scripts/MonoBehaviours/WorkerFlagTest.cs]:
```suggestion
    [WorkerType(WorkerUtils.UnityGameLogic)]
```

File: workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+using UnityEditor;
+using UnityEngine;
+
+#if ODIN_INSPECTOR
+using Sirenix.OdinInspector.Editor;
+#endif
+
+namespace Improbable.Gdk.Debug
+{
+    [CustomEditor(typeof(MonoBehaviour), true)]
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs]: Can we have a named parameter for whatever `true` is?

File: workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+#else
+        protected void OnDisable()
+#endif
+        {
+            if (subscriptions != null)
+            {
+                foreach (var pair in subscriptions)
+                {
+                    var subscription = pair.Value;
+                    subscription?.Cancel();
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs]: Under what conditions should the subscription be null here?

File: workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+                }
+            }
+        }
+
+        private void DrawLiveInspector()
+        {
+            if (isWorkerType.HasValue)
+            {
+                var style = isWorkerType.Value ? enabledContent : disabledContent;
+                style.text = isWorkerType.Value ? workerType : requiredWorkerTypesLabel;
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs]: Nit: `style` is a little misleading imo

File: workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        {
+            return targetType.GetCustomAttribute<WorkerTypeAttribute>()?.WorkerTypes;
+        }
+
+        private static bool HasWorkerTypeAttribute(Type targetType)
+        {
+            return GetRequiredWorkerTypes(targetType) != null;
+        }
+
+        private static bool HasRequireAttributes(Type targetType)
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs]: Can you not use things in the `RequiredSubscriptionsDatabase` for these?

File: workers/unity/Assets/Playground/Scripts/MonoBehaviours/WorkerFlagTest.cs (hunk 1 above)
Comment (zeroZshadow) [reply workers/unity/Assets/Playground/Scripts/MonoBehaviours/WorkerFlagTest.cs]: I thought this was a decent addition for a test? And it shows off the inspector nicely

File: workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs (hunk 3 above)
Comment (zeroZshadow) [reply workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs]: In Editor mode the subscriptions are null

File: workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs (hunk 4 above)
Comment (zeroZshadow) [reply workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs]: `guiContent` ? As its of type `GUIContent`

File: workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs (hunk 3 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs]: Ah right

File: workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs (hunk 4 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs]: Sure 👍

File: workers/unity/Assets/Playground/Scripts/MonoBehaviours/WorkerFlagTest.cs (hunk 1 above)
Comment (jamiebrynes7) [reply workers/unity/Assets/Playground/Scripts/MonoBehaviours/WorkerFlagTest.cs]: Maybe use one of the existing workers instead?

File: workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+                        {
+                            DrawLiveInspector();
+                        }
+                        else
+                        {
+                            DrawEditorInspector();
+                        }
+                    }
+                }
+            }
```
Comment (paulbalaji) [inline workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs]: dat nesting

**Pull request merged**